src/Designer/backend/src/Designer/Services/Implementation/SourceControlSI.cs

```diff
@@ -473,6 +477,133 @@ private async Task CommitAndPushToBranch(string org, string repository, string b
+            }
+
+            RebaseResult rebaseResult = repo.Rebase.Start(
+                repo.Head,
+                upstream,
+                null,
+                identity,
+                rebaseOptions
+            );
+
+
+            if (rebaseResult.Status == RebaseStatus.Conflicts)
+            {
+                repo.Rebase.Abort();
+                throw new InvalidOperationException("Rebase onto latest commit on default branch failed. Rebase aborted.");
+            }
+
+            if (rebaseResult.Status == RebaseStatus.Stop)
+            {
+                repo.Rebase.Abort();
+                throw new InvalidOperationException("Rebase onto latest commit on default branch was stopped by user."); // Should be unreachable code.
+            }
+        }
+
+        /// <inheritdoc/>
+        public void CreateLocalBranch(AltinnRepoEditingContext editingContext, string branchName, string commitSha = null)
+        {
+            string localPath = FindLocalRepoLocation(editingContext);
+            using LibGit2Sharp.Repository repo = new(localPath);
+
+            Branch branch = repo.Branches.FirstOrDefault(branch => branch.FriendlyName == branchName);
+            if (branch is not null) { return; }
+
+            if (commitSha is null)
+            {
+                repo.CreateBranch(branchName);
+                return;
+            }
+            LibGit2Sharp.Commit commit = repo.Lookup<LibGit2Sharp.Commit>(commitSha);
+            repo.CreateBranch(branchName, commit);
+        }
+
+        private static bool LocalBranchExists(LibGit2Sharp.Repository repo, string branchName)
+        {
+            return repo.Branches.Any(branch => branch.FriendlyName == branchName);
+        }
+
+        private static bool LocalBranchIsHead(LibGit2Sharp.Repository repo, string branchName)
+        {
+            return repo.Head.FriendlyName == branchName;
+        }
+
+        /// <inheritdoc/>
+        public void DeleteLocalBranchIfExists(AltinnRepoEditingContext editingContext, string branchName)
+        {
+            string localPath = FindLocalRepoLocation(editingContext);
+            using LibGit2Sharp.Repository repo = new(localPath);
+
+            if (LocalBranchIsHead(repo, branchName))
+            {
+                string defaultBranchName = repo.Branches.Single(branch => branch.FriendlyName == DefaultBranch).FriendlyName;
+                CheckoutRepoOnBranch(editingContext, defaultBranchName);
+            }
+
+            if (LocalBranchExists(repo, branchName) && LocalBranchIsHead(repo, branchName) is false)
+            {
+                repo.Branches.Remove(branchName);
+            }
+        }
+
+        /// <inheritdoc/>
+        public void CheckoutRepoOnBranch(AltinnRepoEditingContext editingContext, string branchName)
+        {
+            string localPath = FindLocalRepoLocation(editingContext);
+            using LibGit2Sharp.Repository repo = new(localPath);
```

Comment:
Disse to linjene gjentar seg flere steder. Hva med å slå dem sammen til en metode?